src/main/java/gtPlusPlus/xmod/gregtech/api/metatileentity/implementations/base/GregtechMeta_MultiBlockBase.java

```diff
@@ -554,7 +556,7 @@ public int canBufferOutputs(ItemStack[] aItemOutputs, FluidStack[] aFluidOutputs
                                     // Update the stack size in the bus storage map.
                                     y.setValue(aOutputBusStack.stackSize + aOutputStack.stackSize);
                                     // Remove the 'input' stack from the recipe outputs, so we don't try count it again.
-                                    aInputMap.remove(u);
+                                    iterator.remove();
                                     continue outputItems;
```

Comment:
There could be a check to continue bus iteration when the stack is full
Also a few lines before (L550) can be moved before stack equal check for some optimization as well.